### ISSUE TITLE
Listener ping for Pub/Sub reconnection, etc.

### ISSUE THREAD

**Issue by ptaoussanis** [ptaoussanis/carmine#15]
Waiting on Redis change: http://goo.gl/LPhIO

**Comment by ptaoussanis** [comment]
Believe this issue has been unblocked with https://github.com/Azure/redis/commit/cfa60a9626896db689de1cbf3b618d57d53f3715 - will need to investigate to confirm.

**Comment by ptaoussanis** [comment]
Update: will be part of Redis 3.2 (no mention of back-porting). 

Carmine'll need updating to use the new feature (should be fairly straightforward).

**Comment by some-long-random-name** [comment]
Any updates on this issue?

**Comment by rwilson** [comment]
There's now support for `PING` in pub/sub contexts, but it's not immediately clear whether this can be effectively solved at the carmine layer.

I considered automatic "reconnect and resubscribe" logic, but while the `subscription-commands` could be saved from listener creation to support resubscribing, I'm unsure how to capture later subscribes using `with-open-listener`.

It might be simpler to more readily _surface_ disconnection. There's basically a mechanism in place; the `listener` includes the future for parsing replies, which becomes realized on disconnect. 

Basically, abnormal disconnect would be: 
```clojure
(and (future-done? (:future listener)) 
     (not (future-cancelled? (:future listener))))
```

So it seems, as is, a caller can await that future and test whether it was a close vs disconnect. It's a bit cumbersome, but doable.

**Comment by ptaoussanis** [comment]
Addressed in upcoming release.
